lualib/resty/kong/peer_conn.lua

```diff
@@ -0,0 +1,39 @@
+local ffi              = require "ffi"
+local base             = require "resty.core.base"
+
+local get_request      = base.get_request
+local errmsg           = base.get_errmsg_ptr()
+local C                = ffi.C
+local ffi_str          = ffi.string
+local get_phase        = ngx.get_phase
+local NGX_ERROR        = ngx.ERROR
+
+local error            = error
+
+
+ffi.cdef[[
+int
+ngx_http_lua_kong_ffi_get_last_peer_connection_cached(ngx_http_request_t *r,
+    char **err);
+]]
+
+
+local function get_last_peer_connection_cached()
+    if get_phase() ~= "balancer" then
+        error("get_last_peer_connection_cached() can only be called in balancer phase")
+    end
+
+    local r = get_request()
```

Comment:
```suggestion
    local r = assert(get_request(), "no request found")
```

Did not meant to remove this important check, just no reason to separate it into a function because nothing else is using it.

Reply:
done